README
Task-SOSA version 1.04



INSTALLATION

To install this module, run the following commands:

    perl Makefile.PL
    make
    make test
    make install


Alternatively, to install with Module::Build, you can use the following commands:

    perl Build.PL
    ./Build
    ./Build test
    ./Build install



DEPENDENCIES

None.


COPYRIGHT AND LICENCE

Copyright (C) 2006 Suretec Systems Ltd., 
              - Gavin Henry <ghenry@suretecsystems.com>
              - http://www.suretecsystems.com

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.
See perldoc perlgpl

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.


Changes
Revision history for Task-SOSA

1.04  The Apr 20 2006 
      - Added Catalyst::View::TT

1.03  Thu Mar 30 2006
      - Version mistake and missing
        Catalyst::Plugin::Authentication::Store::LDAP in docs

1.01  Wed Mar 29 2006
      - Added Catalyst::Plugin::Authentication::Store::LDAP

1.00  Tues Mar 28 2006
      - Updated for new release of Samba-LDAP 
        and bad CPAN upload version number

0.02  Sun Mar 26 2006
      - Updated for new release of Samba-LDAP.

0.01  Fri Mar 24 2006
      - Initial release.



lib/Task/SOSA.pm
package Task::SOSA;

use strict;

our $VERSION = '1.04';

1; # Magic true value required at end of module
__END__

=head1 NAME

Task::SOSA - Install all the CPAN modules needed by SOSA


=head1 VERSION

This document describes Task::SOSA version 1.04


=head1 SYNOPSIS

    perl -MCPAN -e 'install Task::SOSA'
  
=head1 DESCRIPTION

This L<Task> module installs all the dependencies needed by L<SOSA>,
the "Samba and Open-xchange Simple Administrator".

L<Catalyst>

L<Catalyst::Model::LDAP>

L<Catalyst::Plugin::StackTrace>

L<Catalyst::Plugin::DefaultEnd>

L<Catalyst::Plugin::FormValidator>

L<Catalyst::Plugin::FillInForm>

L<Catalyst::Plugin::Authentication>

L<Catalyst::Plugin::Authentication::Store::Minimal>

L<Catalyst::Plugin::Authentication::Credential::Password>

L<Catalyst::Plugin::Authorization::Roles>

L<Catalyst::Plugin::Authorization::ACL>

L<Catalyst::Plugin::Session>

L<Catalyst::Plugin::Session::State::Cookie>

L<Catalyst::Plugin::Session::Store::File>

L<Catalyst::Plugin::Static::Simple>

L<Catalyst::Plugin::Email::Page>

L<Catalyst::Plugin::Prototype>

L<Class::Base>

L<Config::Tiny>

L<Crypt::SmbHash>

L<DateTime>

L<Digest::MD5>

L<Digest::SHA1>

L<List::MoreUtils>

L<MIME::Base64>

L<Net::LDAP>

L<Path::Class>

L<Readonly>

L<Regexp::DefaultFlags>

L<Samba::LDAP>

L<Unicode::MapUTF8>

L<YAML>

See L<http://sosa.sourceforge.net> for more information.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests to
C<bug-task-sosa@rt.cpan.org>, or through the web interface at
L<http://rt.cpan.org>.


=head1 AUTHOR

Gavin Henry  C<< <ghenry@suretecsystems.com> >>

=head1 LICENCE AND COPYRIGHT

Copyright (c) 2006, Suretec Systems Ltd. - Gavin Henry
C<< <ghenry@suretecsystems.com> >>

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version. See L<perlgpl>.


=head1 DISCLAIMER OF WARRANTY

BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT WHEN
OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER
EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH
YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
NECESSARY SERVICING, REPAIR, OR CORRECTION.

IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENCE, BE
LIABLE TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL,
OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE
THE SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF
SUCH DAMAGES.


Makefile.PL
use strict;
use warnings;

# Load the Module::Install bundled in ./inc/
use inc::Module::Install;

# Name of your distribution
name            'Task-SOSA';

# Get most of the details from the primary module
# (The filename argument below is optional if it is placed
# consistently with the "name" above.)
all_from        'lib/Task/SOSA.pm';

# Just in case
license('gpl');

requires( 
    'Catalyst'                          		=> '5.66', 
    'Catalyst::Model::LDAP'             		=> '0.14',
    'Catalyst::Plugin::StackTrace'      		=> '0',
    'Catalyst::Plugin::DefaultEnd'      		=> '0',
    'Catalyst::Plugin::FormValidator'   		=> '0',
    'Catalyst::Plugin::FillInForm'      		=> '0',
    'Catalyst::Plugin::Authentication'  		=> '0',
    'Catalyst::Plugin::Authentication::Store::Minimal'       => '0',
    'Catalyst::Plugin::Authentication::Store::LDAP'          => '0',
    'Catalyst::Plugin::Authentication::Credential::Password' => '0',
    'Catalyst::Plugin::Authorization::Roles'    => '0',
    'Catalyst::Plugin::Authorization::ACL'      => '0',
    'Catalyst::Plugin::Session'                 => '0',
    'Catalyst::Plugin::Session::State::Cookie'  => '0',
    'Catalyst::Plugin::Session::Store::File'    => '0',
    'Catalyst::Plugin::Static::Simple'          => '0',
    'Catalyst::Plugin::Email::Page'             => '0',
    'Catalyst::Plugin::Prototype'               => '0',
    'Catalyst::View::TT'                        => '0',
    'Class::Base'           					=> '0.03',
	'Config::Tiny'          					=> '2.05',
	'Crypt::SmbHash'        					=> '0.12',
 	'DateTime'									=> '0',
	'Digest::MD5'           					=> '2.36',
	'Digest::SHA1'          					=> '2.11',
	'File::Path'								=> '0',
	'List::MoreUtils'       					=> '0.19',
	'MIME::Base64'          					=> '3.07',
	'Net::LDAP'             					=> '0.33',
	'Path::Class'								=> '0',
	'Readonly'              					=> '1.03',
	'Regexp::DefaultFlags'  					=> '0.01',
	'Samba::LDAP'								=> '0.01',
	'Unicode::MapUTF8'      					=> '1.11',
	'UNIVERSAL::require'						=> '0.10',
 	'YAML'										=> '0',
);
    
build_requires  'Test::More'         			=> '0.62';
build_requires  'Test::Distribution' 			=> '1.23';
    
# Auto-install all dependencies from CPAN
auto_install;

# GPG Sign Distribution
sign;

# Generate the Makefile
WriteAll;
